File: app-java/src/main/java/science/monke/internal/order/entity/OrderItemEntity.java
package science.monke.internal.order.entity;

import lombok.Builder;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@Table(value = "order_item")
public class OrderItemEntity {
  @Id private long id;

  @Column(value = "creation_date")
  private OffsetDateTime creationDate;

  @Column(value = "order_id")
  private UUID orderId;

  @Column(value = "item_name")
  private String itemName;
}

File: app-java/src/main/kotlin/science/monke/internal/order/entity/OrderItemEntity.kt
package science.monke.internal.order.entity;

import science.monke.internal.util.AbstractEntity
import java.time.OffsetDateTime
import javax.persistence.*

@Entity
@Table(name = "order_item", schema = "spring_native_reference_project")
class OrderItemEntity(

    @Column(name = "creation_date")
    var creationDate: OffsetDateTime,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", referencedColumnName = "order_id")
    var order: OrderEntity,

    @Column(name = "item_name")
    var itemName: String
) : AbstractEntity<Int>()
